### DISCUSSION TITLE
[Feature idea/request] is possible to mock non m365 API response?

### DISCUSSION THREAD

**Discussion by xzf0587** [dotnet/dev-proxy#1233] [Q&A] [answered]
The guide said it could simulate error for non m365 API. Does it support to mock response for them? 
The scenario is that,
in teams app, we will get auth code in client side and exchange OBO code in backend side before calling graph API.
Usually we will be block on the auth step due to lack of permission or etc.
Is it possible to proxy the AAD auth request to make the simulation flow better?
Thanks.

**Comment by waldekmastykarz** [comment, marked as answer]
Yes, you can use the proxy to simulate any API. Here's an example for OpenAI API for example: https://adoption.microsoft.com/en-us/sample-solution-gallery/sample/pnp-openai-throttling/. Let us know if it's helpful and if you've got any other questions
